src/Adapters/EventSource.Netstandard13/EventSourceTelemetryModule.cs

```diff
@@ -218,5 +230,66 @@ private void EnableAsNecessary(EventSource eventSource)
                 }
             }
         }
+
+        /// <summary>
+        /// Returns true when eventSource satisfied the rule; false otherwise. Returns false when either is null.
+        /// </summary>
+        /// <param name="eventSource">The target event source.</param>
+        /// <param name="rule">The naming rule to be used for matching.</param>
+        private bool IsEventSourceNameMatch(EventSource eventSource, EventSourceNamingMatchRuleBase rule)
+        {
+            if (rule == null || eventSource == null)
+            {
+                return false;
+            }
+
+            if (!rule.IsWildcard)
+            {
+                return string.Equals(eventSource.Name, rule.Name, StringComparison.Ordinal);
+            }
+            else
+            {
+                return IsEventSourceNameMatch(eventSource?.Name, rule?.Name);
+            }
+        }
+
+        /// <summary>
+        /// Returns true when eventSourceName satisfied the rule; false otherwise. Returns false when either is null.
+        /// </summary>
+        /// <param name="eventSourceName"></param>
+        /// <param name="rule"></param>
+        /// <returns></returns>
+        internal bool IsEventSourceNameMatch(string eventSourceName, string rule)
+        {
+            if (string.IsNullOrEmpty(eventSourceName) || string.IsNullOrEmpty(rule))
+            {
+                return false;
+            }
+
+            string pattern = rule.WildCardToRegex();
+            return Regex.IsMatch(eventSourceName, pattern);
```

Comment:
I am concerned about the cost of regex-matching for every event. This what will happen if 10s of thousands of events need to be processed. Even if compiled regular expressions are used (which you do not use here), it is still too costly IMO.

I suggest we switch to using a simple prefix and change the disablement test to string.StartsWith()

Reply:
I think this is a valid concern. I did some quick benchmark:
100,000,000 iterations on match 2 strings - 1 success, 2 fail. And here are the results:

|Method|Stopwatcher Elapsed|
|--|--|
|StartsWith| ~1.27s|
|Contains| ~ 4.7s |
|RegEx| ~ 28.21s |

Based on the result, I think we should refactor the code to use 'StartWith' for partial match.

The configuration shall look like this:
```xml
<Sources>
    <Add Name="MyCompany" Level="Verbose" IsPrefixMatch="false" />
    <Add Name="Microsoft.ApplicationInsights" IsPrefixMatch="true" />
</Sources>
<DisabledSources>
    <Add Name="SomeBadEventSource" IsPrefixMatch="true" />
</DisabledSources>
```

What do you think? @karolz-ms, @SergeyKanzhelev ?

Benchmark code just FYI:
```csharp
using System;
using System.Diagnostics;
using System.Text.RegularExpressions;

namespace RegexVSStartsWithBenchMark
{
    class Program
    {
        private Regex _regex;

        public Program()
        {
            _regex = new Regex("^hello.*", RegexOptions.Compiled);
        }

        static void Main(string[] args)
        {
            const int iterationNumber = 100000000;
            Program p = new Program();
            Stopwatch stopwatch = new Stopwatch();

            Predicate<string> predicate = p.RegExStartWith;
            // Predicate<string> predicate = p.IsStartWith;
            // Predicate<string> predicate = p.Contians;


            stopwatch.Start();
            for (int i = 0; i < iterationNumber; i++)
            {
                predicate("hello");
                predicate("xxxxx");
            }
            stopwatch.Stop();
            Console.WriteLine($"Elapsed: {stopwatch.Elapsed}");
        }
        private bool IsStartWith(string value)
        {
            return value.StartsWith("hello", StringComparison.Ordinal);
        }

        private bool RegExStartWith(string value)
        {
            return _regex.IsMatch(value);
        }

        private bool Contians(string value) {
            return value.Contains("el");
        }
    }
}
```

Reply:
I've updated the implementation to use prefix match for both listening request and disabling requests so that the configuration feels more consistent.
Add sources becomes less flexible than the last iteration but at the same time, at this point, I haven't yet been able to justify that we will have a huge amount of users want using either prefix match or wildcards match for the event source names to enable.